src/ring_theory/polynomial/hermite.lean

```diff
@@ -50,4 +51,71 @@ begin
   simp only [map_one, mul_one, derivative_one, sub_zero]
 end
 
+section coeff
+
+lemma hermite_coeff_recur_zero (n : ℕ) :
+  coeff (hermite (n + 1)) 0 = -(coeff (hermite n) 1) := by simp [coeff_derivative]
+
+lemma hermite_coeff_recur (n k : ℕ) :
+  coeff (hermite (n + 1)) (k + 1) = coeff (hermite n) k - (k + 2) * (coeff (hermite n) (k + 2)) :=
+begin
+  rw [hermite_succ, coeff_sub, coeff_X_mul, coeff_derivative, mul_comm],
+  norm_cast
+end
+
+lemma hermite_coeff_upper (n k : ℕ) : coeff (hermite n) (n + k + 1) = 0 :=
+begin
+  induction n with n ih generalizing k,
+  { apply coeff_C },
+  {  rw [hermite_coeff_recur,
+      (by linarith : n + 1 + k = n + k + 1),
+      (by linarith : n + k + 1 + 2 = n + (k + 2) + 1),
+      ih k, ih (k + 2), mul_zero, sub_zero] }
+end
+
+lemma hermite_coeff_upper' (n k : ℕ) (hnk : n < k) : coeff (hermite n) k = 0 :=
+begin
+  obtain ⟨k, rfl⟩ := nat.exists_eq_add_of_lt hnk,
+  apply hermite_coeff_upper
+end
+
+lemma hermite_coeff_top (n : ℕ) : coeff (hermite n) n = 1 :=
+begin
+  induction n with n ih,
+  { apply coeff_C },
+  { rw [hermite_coeff_recur, ih, hermite_coeff_upper, mul_zero, sub_zero] }
+end
+
+lemma hermite_degree {n : ℕ} : (hermite n).degree = n :=
+begin
+  rw degree_eq_of_le_of_coeff_ne_zero,
+  simp_rw [degree_le_iff_coeff_zero, with_bot.coe_lt_coe],
+  exact hermite_coeff_upper' n,
+  simp [hermite_coeff_top n]
+end
+
+lemma hermite_nat_degree {n : ℕ} : (hermite n).nat_degree = n :=
+nat_degree_eq_of_degree_eq_some hermite_degree
+
+lemma hermite_leading_coeff {n : ℕ} : (hermite n).leading_coeff = 1 :=
+begin
+  rw [← coeff_nat_degree, hermite_nat_degree, hermite_coeff_top],
+end
+
+lemma hermite_monic {n : ℕ} : (hermite n).monic := hermite_leading_coeff
+
+lemma hermite_coeff_parity_zero (n k : ℕ) (hnk : odd (n + k)) : coeff (hermite n) k = 0 :=
```

Comment:
This is the one place where `n` should be implicit, since it can be inferred from `hnk`
```suggestion
lemma coeff_hermite_of_odd_add {n k : ℕ} (hnk : odd (n + k)) : coeff (hermite n) k = 0 :=
```

Reply:
I don't think `n` and `k` can both be inferred simultaneously from `hnk`.

Reply:
I would expect them to be able to; the current proof failure is unrelated